lib/service/connect.go

```diff
@@ -1059,21 +1082,23 @@ func (process *TeleportProcess) newClient(authServers []utils.NetAddr, identity
 
 	logger.Debug("Attempting to discover reverse tunnel address.")
 
-	logger.Debug("Attempting to connect to Auth Server through tunnel.")
-	sshClientConfig, err := identity.SSHClientConfig(process.Config.FIPS)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
+	// TODO: Remove tunneling to a proxy here by v14
+	logger.Debug("Attempting to connect to Proxy Server through tunnel.")
```

Comment:
If you can determine that the config version is v3, you can skip dialing via tunnel here. V3 configs will be opt-in, so we can make proxy_servers required for tunneling.

Reply:
Ah that `TODO` is an old comment that can be removed. This tunnelling is needed as it's the fallback after trying to direct dial the value from `AuthServers`.

I've made it so v3 configs are default - is there a reason we'd make it opt-in? The way I saw it was there's no backwards compatibility issues, so we can skip straight to v3 being the default.

Reply:
We don't know the config version in this function to know to make v3 _only_ direct dial an auth server or tunnel a proxy server. We can guess, which is what I'm doing now - if proxy addresses are specified, tunnel to them, if not, direct dial to the auth server value and fall back to tunnelling if it doesn't work. 

Maybe it would be nicer to pass the configuration version to here, so we can restrict the fallback of tunnelling the auth addresses value to v2 and below.